OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -684,8 +684,11 @@ string GetSourceDisplayName(IReadOnlyPackage source)
 					name = "^" + name[Platform.SupportDir.Length..];
 			}
 
-			if (name.Length > 18)
-				name = "..." + name[^15..];
+			if (!ChromeMetrics.TryGet<int>("AssetBrowserSourceMaxLength", out var maxLength))
```

Comment:
This is quite obscure, wouldn't it be better solution be something like `WidgetUtils.TruncateLabelToTooltip` does?

Or instead have maxLength be exposed in the Widget itself (overwritten via yaml)? I wouldn't mind applying both solutions, we are slowly moving towards adding tooltips to all truncated labels.

Reply:
Tooltips are also nice, but the problem here is that regardless of that, the logic here cuts to a specific length, without knowing how much character will fit in the actual asset browser. So i see 2 solutions here

- Keep it asset browser specific. Then we need a chrome metric, as i cannot add parameters to logics.
- Remove this hardcoded thing. Then overhaul how dropdowns generally work, so they automatically cut their containing text depending on their size.

The second approach might actualy have some heavier side effects on all usages of dropdowns. Especialy as its possible they not only contain text after all.

Got some further feedback on this?